src/sentry/snuba/metrics/extraction.py

```diff
@@ -586,6 +608,11 @@ def should_use_on_demand_metrics(
         return False
 
     function, args = components
+
+    # This helps us control which functions are allowed to use the new spec version.
+    if query_builder_call and function in NEXT_SPEC_VERSION_FUNCTIONS and not use_next_spec_version:
+        return False
```

Comment:
This line should have coverage. I will check again when all the results have come in.